### PR TITLE
add n17994

### PR DESCRIPTION
**Pull request by vasild**

### CODE REVIEW

File: _posts/2020-04-22-#17994.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+- file [fsync](https://linux.die.net/man/2/fsync) - moving all modified data from the OS buffers to the disk.
+
+## Questions
+
+### Overview
+
+- How/when is the blocks database (blocks/{blk,rev}\*.dat) created?
+- Can we create an undo for a given block without having all prior blocks?
+- Do we ever modify existent data in the blocks database or we just append new blocks and their undos?
```
Comment (jonatack) [inline _posts/2020-04-22-#17994.md:35]: s/existent/existing/ and s/or/,or do/

File: _posts/2020-04-22-#17994.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+host: vasild
+status: upcoming
+commit:
+---
+
+## Notes
+
+In this Review Club we will dive into how blocks are downloaded and stored on disk. How disk access is arranged and a bug it has caused.
+
+Some concepts that would help to grasp what's going on:
```
Comment (jonatack) [inline _posts/2020-04-22-#17994.md:17]: Suggestion: consider replacing this sentence with a "### Concepts" subheader, e.g.

### Concepts

File: _posts/2020-04-22-#17994.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+- [Initial Block Download](https://bitcoin.org/en/glossary/initial-block-download)
+
+- "block undo" - all the information that is necessary to revert a block, if a chain reorganization happens, for example.
+
+- [class FlatFileSeq](https://doxygen.bitcoincore.org/class_flat_file_seq.html)
```
Comment (jonatack) [inline _posts/2020-04-22-#17994.md:23]: could you add a sentence of context about what is notable about this class?

File: _posts/2020-04-22-#17994.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+### Overview
+
+- How/when is the blocks database (blocks/{blk,rev}\*.dat) created?
+- Can we create an undo for a given block without having all prior blocks?
+- Do we ever modify existent data in the blocks database or we just append new blocks and their undos?
+
+### Dive
+
+- How is space allocated in the files when new data is appended? Why?
+- What does to "finalize" a file mean?
```
Comment (jonatack) [inline _posts/2020-04-22-#17994.md:40]: What does it mean to "finalize" a file?

File: _posts/2020-04-22-#17994.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- What is the bug that the PR is fixing?
+- How would you demonstrate the bug existence?
```
Comment (jonatack) [inline _posts/2020-04-22-#17994.md:42]: How would you reproduce the bug or show it exists?

File: _posts/2020-04-22-#17994.md (hunk 1 above)
Comment (vasild) [reply _posts/2020-04-22-#17994.md]: Done

File: _posts/2020-04-22-#17994.md (hunk 2 above)
Comment (vasild) [reply _posts/2020-04-22-#17994.md]: Done, except I used 3 `###`, not 4.

File: _posts/2020-04-22-#17994.md (hunk 3 above)
Comment (vasild) [reply _posts/2020-04-22-#17994.md]: Done

File: _posts/2020-04-22-#17994.md (hunk 5 above)
Comment (vasild) [reply _posts/2020-04-22-#17994.md]: Done

File: _posts/2020-04-22-#17994.md (hunk 4 above)
Comment (vasild) [reply _posts/2020-04-22-#17994.md]: Done

**Comment by vasild** [comment]
Updated. Render preview: https://github.com/bitcoin-core-review-club/website/blob/9f5229f/_posts/2020-04-22-%2317994.md

In order to extend the `Notes` section I had to sacrifice the first question `How/when is the blocks database (blocks/{blk,rev}\*.dat) created?` and answer it there :)

**Comment by jnewbery** [comment]
Thanks for this Vasil. Looks great. I've made a few changes (mostly formatting and style).

**Pull request merged**